addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1361,3 +1412,108 @@ func (r *ClusterBootstrapReconciler) reconcileClusterProxyAndNetworkSettings(clu
 
 	return nil
 }
+
+func (r *ClusterBootstrapReconciler) reconcileDelete(cluster *clusterapiv1beta1.Cluster, log logr.Logger) (ctrl.Result, error) {
+	remoteClient, err := util.GetClusterClient(r.context, r.Client, r.Scheme, clusterapiutil.ObjectKey(cluster))
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
+	clusterBootstrap := &runtanzuv1alpha3.ClusterBootstrap{}
+	err = remoteClient.Get(r.context, client.ObjectKeyFromObject(cluster), clusterBootstrap)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if clusterBootstrap == nil {
+		return ctrl.Result{}, nil
+	}
+
+	err = r.removeAdditionalPackageInstalls(cluster, log)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	packageInstallsArePresent := hasPackageInstalls(r.context, remoteClient, cluster, r.Config.SystemNamespace, clusterBootstrap.Spec.AdditionalPackages, log)
+	timeOutReached := time.Now().After(cluster.GetDeletionTimestamp().Add(clusterDeleteTimeout))
+
+	if packageInstallsArePresent || timeOutReached {
```

Comment:
If `hasPackageInstalls`  is wrapped in a retry look, we probably don't need timeOutReached.

Reply:
typo: retry look -> retry loop

Reply:
Reworked the logic so as to only make calls to api itimeoutReached = false.